File: library/actor-common/src/main/java/im/actor/model/entity/MessageState.java
package im.actor.model.entity;

/**
 * Created by ex3ndr on 09.02.15.
 */
public enum MessageState {
    PENDING(1),
    SENT(2),
    RECEIVED(3),
    READ(4),
    ERROR(5),
    UNKNOWN(6);

    int value;

    MessageState(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static MessageState fromValue(int value) {
        switch (value) {
            case 1:
                return PENDING;
            case 2:
                return SENT;
            case 3:
                return RECEIVED;
            case 4:
                return READ;
            case 5:
                return ERROR;
            default:
            case 6:
                return UNKNOWN;
        }
    }
}

File: actor-server/actor-models/src/main/scala/im/actor/server/model/MessageState.scala
package im.actor.server.model

sealed trait MessageState {
  def toInt: Int
}

object MessageState {
  @SerialVersionUID(1L)
  case object Sent extends MessageState {
    def toInt = 1
  }

  @SerialVersionUID(1L)
  case object Received extends MessageState {
    def toInt = 2
  }

  @SerialVersionUID(1L)
  case object Read extends MessageState {
    def toInt = 3
  }

  def fromInt(i: Int): MessageState = i match {
    case 1 ⇒ Sent
    case 2 ⇒ Received
    case 3 ⇒ Read
  }
}
